core/logger/audit/audit_logger.go

```diff
@@ -0,0 +1,230 @@
+package audit
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+
+	"github.com/pkg/errors"
+)
+
+const bufferCapacity = 2048
+
+type Data = map[string]any
+
+type AuditLogger interface {
+	Audit(eventID EventID, data Data)
+}
+
+type AuditLoggerConfig struct {
+	forwardToUrl   string
+	environment    string
+	jsonWrapperKey string
+	headers        []serviceHeader
```

Comment:
For this to be directly compatible with the new toml config, it should use capitalized field names, and pointer types. `models.URL` is helpful too.
```suggestion
	ForwardToUrl   *models.URL
	Environment    *string
	JsonWrapperKey *string
	Headers        []serviceHeader
```